Translation of "Building on Lisk" into Indonesian (#264)
LiskHQ/lisk-docs: i18n/ind/docusaurus-plugin-content-docs/current/building-on-lisk/deploying-smart-contract/with-thirdweb.mdx

---
title: ... menggunakan thirdweb
slug: /building-on-lisk/deploying-smart-contract/with-thirdweb

> description: "Panduan untuk mendeploy smart contract di jaringan test Lisk menggunakan thirdweb. Termasuk instruksi untuk mengatur environment, mengompilasi, dan mendeploy smart contract."

Reviewer: ```suggestion
description: "Panduan untuk meluncurkan kontrak pintar di jaringan test Lisk menggunakan thirdweb. Termasuk instruksi untuk mengatur environment, mengompilasi, dan meluncurkan kontrak pintar."
```

keywords: [
    "thirdweb",

>     "smart contract",

Reviewer: ```suggestion
    "kontrak pintar",
```

Lebih banyak informasi tentang ini dapat ditemukan di bagian [thirdweb Explore](#meluncurkan-kontrak-bawaan-melalui-thirdweb-explore).

Selain itu, developer yang ingin mengembangkan dan meluncurkan kontrak pintar kustom dapat menggunakan [Solidity SDK](https://portal.thirdweb.com/contracts/build/overview) yang disediakan oleh thirdweb, serta opsi lainnya seperti [TypeScript SDK](https://portal.thirdweb.com/typescript/v5) dan [Unity SDK](https://portal.thirdweb.com/unity).  
Namun, dalam panduan ini, kami akan lebih fokus pada [thirdweb Explore](https://thirdweb.com/explore) dan [Solidity SDK](https://portal.thirdweb.com/contracts/build/overview).

**Fitur:**

- [Explore](https://portal.thirdweb.com/contracts/explore/overview): Meluncurkan kontrak bawaan siap pakai.
- [Build (Solidity SDK)](https://portal.thirdweb.com/contracts/build/overview): Menulis kontrak pintar kustom.
- [Deploy](https://portal.thirdweb.com/contracts/deploy/overview): Mendukung peluncuran kontrak untuk berbagai kasus penggunaan.
- [Publish](https://portal.thirdweb.com/contracts/publish/overview): Mempublikasikan kontrak Anda secara on-chain.

> - [Interact](https://portal.thirdweb.com/contracts/interact/overview): Berinteraksi dengan kontrak pintar dan mengintegrasikannya langsung ke dalam aplikasi Anda.

Reviewer 2: ```suggestion
- Interact: Berinteraksi dengan kontrak pintar dan mengintegrasikannya langsung ke dalam aplikasi Anda.
```

Lebih banyak informasi tentang ini dapat ditemukan di bagian [thirdweb Explore](#deploy-kontrak-bawaan-melalui-thirdweb-explore).

Selain itu, developer yang ingin mengembangkan dan deploy smart contract kustom dapat menggunakan [Solidity SDK](https://portal.thirdweb.com/contracts/build/overview) yang disediakan oleh thirdweb, serta opsi lainnya seperti [TypeScript SDK](https://portal.thirdweb.com/typescript/v5) dan [Unity SDK](https://portal.thirdweb.com/unity).  
Namun, dalam panduan ini, kami akan lebih fokus pada [thirdweb Explore](https://thirdweb.com/explore) dan [Solidity SDK](https://portal.thirdweb.com/contracts/build/overview).

**Fitur:**

- [Explore](https://portal.thirdweb.com/contracts/explore/overview): Deploy kontrak bawaan siap pakai.
- [Build (Solidity SDK)](https://portal.thirdweb.com/contracts/build/overview): Menulis smart contract kustom.
- [Deploy](https://portal.thirdweb.com/contracts/deploy/overview): Mendukung deployment kontrak untuk berbagai kasus penggunaan.
- [Publish](https://portal.thirdweb.com/contracts/publish/overview): Mempublikasikan kontrak Anda secara on-chain.

> - Interact: Berinteraksi dengan smart contract dan mengintegrasikannya langsung ke dalam aplikasi Anda.

Reviewer 2: ```suggestion
```

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> # Mendeploy Smart Contract dengan thirdweb

Reviewer: ```suggestion
# Meluncurkan Kontrak Pintar dengan thirdweb
```

> [thirdweb](https://thirdweb.com/) adalah _framework_ end-to-end untuk pengembangan dan deployment _smart contract_.

Reviewer: ```suggestion
[thirdweb](https://thirdweb.com/) adalah _framework_ end-to-end untuk pengembangan dan peluncuran kontrak pintar.
```

> Kerangka ini memungkinkan developer untuk mendeploy kontrak standar seperti ERC-20, ERC-721, atau ERC-1155, tanpa menulis satu baris kode pun.  

Reviewer: ```suggestion
Kerangka ini memungkinkan developer untuk meluncurkan kontrak standar seperti ERC-20, ERC-721, atau ERC-1155, tanpa menulis satu baris kode pun.  
```

Lebih banyak informasi tentang ini dapat ditemukan di bagian [thirdweb Explore](#mendeploy-kontrak-bawaan-melalui-thirdweb-explore).

> Selain itu, developer yang ingin mengembangkan dan mendeploy _smart contract_ kustom dapat menggunakan [Solidity SDK](https://portal.thirdweb.com/contracts/build/overview) yang disediakan oleh thirdweb, serta opsi lainnya seperti [TypeScript SDK](https://portal.thirdweb.com/typescript/v5) dan [Unity SDK](https://portal.thirdweb.com/unity).  

Reviewer: ```suggestion
Selain itu, developer yang ingin mengembangkan dan meluncurkan kontrak pintar kustom dapat menggunakan [Solidity SDK](https://portal.thirdweb.com/contracts/build/overview) yang disediakan oleh thirdweb, serta opsi lainnya seperti [TypeScript SDK](https://portal.thirdweb.com/typescript/v5) dan [Unity SDK](https://portal.thirdweb.com/unity).  
```

Namun, dalam panduan ini, kami akan lebih fokus pada [thirdweb Explore](https://thirdweb.com/explore) dan [Solidity SDK](https://portal.thirdweb.com/contracts/build/overview).

**Fitur:**

> - [Explore](https://portal.thirdweb.com/contracts/explore/overview): Mendeploy kontrak bawaan siap pakai.

Reviewer: ```suggestion
- [Explore](https://portal.thirdweb.com/contracts/explore/overview): Meluncurkan kontrak bawaan siap pakai.
```

> - [Build (Solidity SDK)](https://portal.thirdweb.com/contracts/build/overview): Menulis _smart contract_ kustom.

Reviewer: ```suggestion
- [Build (Solidity SDK)](https://portal.thirdweb.com/contracts/build/overview): Menulis kontrak pintar kustom.
```

> - [Deploy](https://portal.thirdweb.com/contracts/deploy/overview): Mendukung deployment kontrak untuk berbagai kasus penggunaan.

Reviewer: ```suggestion
- [Deploy](https://portal.thirdweb.com/contracts/deploy/overview): Mendukung peluncuran kontrak untuk berbagai kasus penggunaan.
```

- [Publish](https://portal.thirdweb.com/contracts/publish/overview): Mempublikasikan kontrak Anda secara on-chain.

> - [Interact](https://portal.thirdweb.com/contracts/interact/overview): Berinteraksi dengan _smart contract_ dan mengintegrasikannya langsung ke dalam aplikasi Anda.

Reviewer: ```suggestion
- [Interact](https://portal.thirdweb.com/contracts/interact/overview): Berinteraksi dengan kontrak pintar dan mengintegrasikannya langsung ke dalam aplikasi Anda.
```

> ## Mendeploy Kontrak Bawaan melalui thirdweb Explore

Reviewer: ```suggestion
## Meluncurkan Kontrak Bawaan melalui thirdweb Explore
```

> Jika Anda ingin mendeploy kontrak bawaan tanpa kustomisasi, thirdweb menyediakan cara yang mudah untuk melakukannya menggunakan **Explore**.  

Reviewer: ```suggestion
Jika Anda ingin meluncurkan kontrak bawaan tanpa kustomisasi, thirdweb menyediakan cara yang mudah untuk melakukannya menggunakan **Explore**.  
```

Tonton video berikut untuk pengenalan singkat tentang Explore:

- [Apa itu thirdweb **Explore**?](https://youtu.be/jYEqoIeAoBg?t=5300)

:::note

> Sebelum mencoba panduan berikut, pastikan bahwa [wallet Anda terhubung ke Jaringan Lisk](/user/connecting-to-a-wallet) dan memiliki dana yang cukup.  

Reviewer: ```suggestion
Sebelum mencoba panduan berikut, pastikan bahwa [dompet Anda terhubung ke Jaringan Lisk](/user/connecting-to-a-wallet) dan memiliki dana yang cukup.  
```

> Untuk informasi lebih lanjut, lihat bagian [Dana Wallet](/building-on-lisk/deploying-smart-contract/with-Foundry#dana-wallet).

Reviewer: ```suggestion
Untuk informasi lebih lanjut, lihat bagian [Dana Dompet](/building-on-lisk/deploying-smart-contract/with-Foundry#dana-dompet).
```

:::

> Video-video berikut menjelaskan langkah-langkah untuk mendeploy berbagai token dengan kontrak bawaan di Lisk:

Reviewer: ```suggestion
Video-video berikut menjelaskan langkah-langkah untuk meluncurkan berbagai token dengan kontrak bawaan di Lisk:
```

> - [Cara mendeploy kontrak bawaan **ERC-20** melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=5776)

Reviewer: ```suggestion
- [Bagaimana cara meluncurkan kontrak bawaan **ERC-20** melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=5776)
```

> - [Cara mendeploy kontrak bawaan **ERC-721** melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=6040)

Reviewer: ```suggestion
- [Bagaimana cara meluncurkan kontrak bawaan **ERC-721** melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=6040)
```

> - [Cara mendeploy kontrak bawaan **ERC-1155** melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=6396)

Reviewer: ```suggestion
- [Bagaimana cara meluncurkan kontrak bawaan **ERC-1155** melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=6396)
```

## Menggunakan Solidity SDK

### Prasyarat
#### Node.js v18+

_Download_ dan instal [Node v18+](https://nodejs.org/en/download/).  

Jika Anda menggunakan `nvm` untuk mengelola versi Node.js Anda, cukup jalankan perintah: `nvm install 18`.

#### Menginstal dan Mengonfigurasi thirdweb

> Untuk mengembangkan kontrak kustom, Anda perlu mengunduh dan mengonfigurasi thirdweb.  

Reviewer: ```suggestion
Untuk mengembangkan kontrak kustom, Anda perlu men_download_ dan mengonfigurasi thirdweb.  
```

    ⠙ Waiting for a response from the dashboard
    ```
3. Setelah Anda melakukan autentikasi melalui browser, terminal akan mencatat pesan berikut:  
    ```text
    Successfully linked your account to this device 
    ```
4. Anda dapat memverifikasi status login Anda dengan menjalankan kembali perintah `thirdweb login`.
    ```text
    💎 thirdweb v0.13.60 💎

### Membuat Proyek

> Langkah pertama untuk mendeploy _smart contract_ ke Lisk adalah menyiapkan _environment_ pengembangan Anda dengan membuat sebuah proyek.  

Reviewer: ```suggestion
Langkah pertama untuk meluncurkan kontrak pintar ke Lisk adalah menyiapkan _environment_ pengembangan Anda dengan membuat sebuah proyek.  
```

> thirdweb mendukung dua framework pengembangan _smart contract_ paling populer, yaitu [Foundry](https://book.getfoundry.sh/) dan [Hardhat](https://hardhat.org).

Reviewer: ```suggestion
thirdweb mendukung dua framework pengembangan kontrak pintar paling populer, yaitu [Foundry](https://book.getfoundry.sh/) dan [Hardhat](https://hardhat.org).
```

> Dalam panduan ini, kami menggunakan Foundry untuk pengembangan _smart contract_.  

Reviewer: ```suggestion
Dalam panduan ini, kami menggunakan Foundry untuk pengembangan kontrak pintar.  
```

    ✔ What type of project do you want to create? › Contract
    ✔ What is your project named? … thirdweb-contracts
    ``` 

> 4. Selanjutnya, pilih `Forge` sebagai framework, `NFT` sebagai nama kontrak, `ERC721` sebagai jenis _smart contract_, dan `None` untuk opsi ekstensi.

Reviewer: ```suggestion
4. Selanjutnya, pilih `Forge` sebagai framework, `NFT` sebagai nama kontrak, `ERC721` sebagai jenis kontrak pintar, dan `None` untuk opsi ekstensi.
```

    └── README.md
    ``` 

> #### **Membuat** smart contract

Reviewer: ```suggestion
#### **Membuat** Kontrak Pintar
```

```

> #### **Mem-build** smart contract

Reviewer: ```suggestion
#### **Membangun** Kontrak Pintar
```

> Setelah kode _smart contract_ siap, kontrak tersebut harus di-_build_ menggunakan thirdweb.

Reviewer: ```suggestion
Setelah kode kontrak pintar siap, kontrak tersebut harus di-_build_ menggunakan thirdweb.
```

1. Untuk melakukannya, cukup jalankan perintah berikut:  
    ```bash
    npx thirdweb build
    ```

> 2. Jika _smart contract_ tidak memiliki error, Anda akan melihat output berikut di terminal:  

Reviewer: ```suggestion
2. Jika kontrak pintar tidak memiliki error, Anda akan melihat output berikut di terminal:  
```

     yarn deploy
    ```

> #### **Mendeploy** smart contract

Reviewer: ```suggestion
#### **Meluncurkan** Kontrak Pintar
```

> Semua pemeriksaan berhasil dilewati, yang berarti _smart contract_ siap untuk dideploy menggunakan thirdweb.

Reviewer: ```suggestion
Semua pemeriksaan berhasil dilewati, yang berarti kontrak pintar siap untuk diluncurkan menggunakan thirdweb.
```

1. Untuk melakukannya, cukup jalankan perintah berikut:
    ```bash
    npx thirdweb deploy
    ```

> 2. Jika _smart contract_ tidak memiliki error, Anda akan melihat output berikut di terminal:

Reviewer: ```suggestion
2. Jika kontrak pintar tidak memiliki error, Anda akan melihat output berikut di terminal:
```

    ✔ Open this link to deploy your contracts: https://thirdweb.com/contracts/deploy/QmSJExQJfPYFuaRZuDu9XRR2jUu9yp3kaFX3Sdc1KRWxiP
    ```

3. Terminal juga akan membuka aplikasi browser yang mengarah ke URL khusus, seperti yang ter-_highlight_ diatas.  
4. Isi formulir sesuai dengan petunjuk di bagian [Fill Parameter](https://portal.thirdweb.com/contracts/deploy/deploy-contract#Fill%20parameter).  

> 5. Setelah Anda mengisi detail _smart contract_ ERC721, klik tombol `Deploy Now` dan kontrak tersebut akan dideploy ke jaringan yang dipilih dari _wallet_ Anda.  

Reviewer: ```suggestion
5. Setelah Anda mengisi detail kontrak pintar ERC721, klik tombol `Deploy Now` dan kontrak tersebut akan diluncurkan ke jaringan yang dipilih dari dompet Anda.  
```

:::info

> Jika Anda ingin mengetes kontrak berbasis Foundry Anda, ikuti langkah-langkah yang disebutkan dalam panduan [Testing the Smart Contract](/building-on-lisk/deploying-smart-contract/with-Foundry#menge-test-smart-contract).

Reviewer: ```suggestion
Jika Anda ingin menguji kontrak berbasis Foundry Anda, ikuti langkah-langkah yang disebutkan dalam panduan [Menguji Kontrak Pintar](/building-on-lisk/deploying-smart-contract/with-Foundry#menge-test-smart-contract).
```

:::

> ## Berinteraksi dengan Smart Contract

Reviewer: ```suggestion
## Berinteraksi dengan Kontrak Pintar
```

> Setelah kontrak dideploy, Anda dapat berinteraksi dengan kontrak tersebut melalui Explore.  

Reviewer: ```suggestion
Setelah kontrak diluncurkan, Anda dapat berinteraksi dengan kontrak tersebut melalui Explore.  
```

Video berikut menjelaskan langkah demi langkah bagaimana berinteraksi dengan bermacam-macam kontrak token di Lisk.

> - [Bagaimana cara berinteraksi dengan kontrak **ERC-20** yang telah dideploy melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=6823)

Reviewer: ```suggestion
- [Bagaimana cara berinteraksi dengan kontrak **ERC-20** yang telah diluncurkan melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=6823)
```

> - [Bagaimana cara berinteraksi dengan kontrak **ERC-721** yang telah dideploy melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=7581)

Reviewer: ```suggestion
- [Bagaimana cara berinteraksi dengan kontrak **ERC-721** yang telah diluncurkan melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=7581)
```

> - [Bagaimana cara berinteraksi dengan kontrak **ERC-1155** yang telah dideploy melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=8355)

Reviewer: ```suggestion
- [Bagaimana cara berinteraksi dengan kontrak **ERC-1155** yang telah diluncurkan melalui Explore?](https://youtu.be/jYEqoIeAoBg?t=8355)
```
